bento_beacon/utils/beacon_request.py

```diff
@@ -192,5 +194,40 @@ def summary_stats_requested():
 
 
 async def verify_permissions():
-    # can do much more here in the future
-    g.permission_query_data = await check_permission(P_QUERY_DATA)
+    resource = requested_resource()
+    permissions = await retrieve_permissions(resource)
+    check_permissions_sufficient_for_request(permissions, resource)
+
+
+def check_permissions_sufficient_for_request(permissions: PermissionsDict, resource: dict) -> None:
+    # permissions are for the resource being requested
+    # so we don't need to re-verify which resource is being requested
+    # but we do need to know whether we are at dataset level or project level
+    is_dataset_level = True if resource.get("dataset") else False
+
+    requested_granularity = g.request_data["requestedGranularity"]
+
+    if requested_granularity == GRANULARITY_RECORD and not has_full_record_permissions(permissions):
```

Comment:
not covered by tests; maybe possible to test the verify_permissions with request mocks or (a more simple version) just write unit tests for this function?